src/util/cracklib-format

```diff
@@ -3,8 +3,16 @@
 # This preprocesses a set of word lists into a suitable form for input
 # into cracklib-packer
 #
+# Truncates lines longer than 1022 characters long as cracklib-packer
+# does not handle them correctly.
+#
+# Uses 'grep -v' to remove any blank lines as cracklib-packer will
+# generate "skipping line" warnings otherwise.
+#
 gzip -cdf "$@" |
     grep -a -v '^\(#\|$\)' |
```

Comment:
this already is doing a `grep -v` to remove blank lines.  why isn't this working ?

Reply:
It is not working as expected because the blank line(s) in question do not exist in the input to that grep command. Rather, the blanks lines are created by the subsequent "tr -cd '\012[a-z][0-9]'" command acting on any line(s) which contain just a single carriage return character (octal 13) which the "tr" will remove, so leaving a blank line. At least one of the word files in the repo does contain at least a single line which is just a single carriage return.

The blank line removal should be the very last part of the pipeline so that if any of the preceeding parts of the pipeline themselves create a blank line it will then be removed.

Line 13's grep could be moved to the end but I will instead rewrite line 13 to just remove any comment lines (no point passing comments to rest of the pipeline and then removing them at the end of the pipeline) and leave the new grep in place at the end to mop up any blank lines.